### ISSUE TITLE
Cannot save the workflow after renaming the condition within the Scope in New designer

### ISSUE THREAD

**Issue by EnochZZ717** [Azure/LogicAppsUX#6086]
### Describe the Bug with repro steps 

- Deployed using cx's template 
- Add a Scope 
- Add a Condition 
- Rename the condition and click save 

![Image](<https://github.com/user-attachments/assets/fd9f6327-d673-4c9b-8017-4d7bf1c7ed16>)

\*\*The issue does not exist in the legacy designer. \*\*

### What type of Logic App Is this happening in? 

Consumption (Portal)

### Which operating system are you using? 

Windows

### Are you using new designer or old designer 

New Designer

### Did you refer to the TSG before filing this issue? [https://aka.ms/lauxtsg](<https://aka.ms/lauxtsg>) 

Yes

### Workflow JSON 

```json
the json is to large, please check the template.json on additional context
```

### Screenshots or Videos

[https://github.com/user-attachments/assets/690e5a73-62ee-4418-bbda-6ac6ac330c1c](<https://github.com/user-attachments/assets/690e5a73-62ee-4418-bbda-6ac6ac330c1c>)

### Browser

Edge

### Additional context

[template.json](<https://github.com/user-attachments/files/17728987/template.json>)

**Comment by ccastrotrejo** [comment]
Hi @EnochZZ717 thanks for raising this issue. It seems that the action that is preventing to save the workflow is the `Loop_until_Completion` action and the expression stated in the until is the one causing the issue.  There are a couple of things to notice over here.

- I can see that the template has the following expression `"expression":"@equals(variables('ConditionPassed'), true)"`
- I can see that the error states that the expression is `equals(variables('ConditionPassed'), \"true\")`
- Could you please check in codeview that the expression is as the template? That should be the correct pattern.
- By any chance d you updated the code from the designer or from codeview? I played around with the condition expression component but couldn't face an scenario in which the statement was changed as the error state

**Comment by EnochZZ717** [comment]
Hi @ccastrotrejo , thanks for your reply. I have checked the Loop_until_Completion action in codeview and it seems there are no issues. You could also check it in template.json. The problem is strange; the error only occurs when we try to rename the condition action within the scope. Please check the recording I just uploaded.
https://github.com/user-attachments/assets/ad1fedbe-1452-476f-b22f-c05206404e06

**Comment by ccastrotrejo** [comment]
Hi @EnochZZ717 thanks for the video, it does sems that for some resaon updating the name alters the content of the loop condition. I will try to repro it

**Comment by EnochZZ717** [comment]
> Hi [@EnochZZ717](https://github.com/EnochZZ717) thanks for the video, it does seems that for some reason updating the name alters the content of the loop condition. I will try to repro it.
> 
> Could you please try and use this private azure url to see if this also happens? https://aka.ms/emastaging

Hi @ccastrotrejo , the issue also happens in this private azure url https://aka.ms/emastaging

**Comment by hartra344** [comment]
I'm not able to repro this issue. @ccastrotrejo  or @EnochZZ717  can one of you confirm this still repros?

**Comment by ccastrotrejo** [comment]
Yes, I’m currently unable to reproduce this issue, so I’ll go ahead and close it for now. I’ll continue to monitor in case it resurfaces.

@EnochZZ717  Could you please try again? If the issue persists, let us know and we’ll take another look.
